dhcpv4/nclient4/client.go

```diff
@@ -249,54 +303,57 @@ func (c *Client) receiveLoop() {
 }
 
 // ClientOpt is a function that configures the Client.
-type ClientOpt func(*Client)
+type ClientOpt func(c *Client) error
 
 // WithTimeout configures the retransmission timeout.
 //
 // Default is 5 seconds.
 func WithTimeout(d time.Duration) ClientOpt {
-	return func(c *Client) {
+	return func(c *Client) (err error) {
 		c.timeout = d
+		return
 	}
 }
 
-// WithSummaryLogger logs one-line DHCPv4 message summarys when sent & received.
-func WithSummaryLogger() ClientOpt {
```

Comment:
Please keep the WithSummaryLogger and WithDebugLogger for people already using nclient4.